Chainbuilder `TestRun` flake

## Context

https://github.com/celestiaorg/celestia-app/actions/runs/14685468484/job/41213492847?pr=4672#step:4:778

## Problem

```
=== RUN   TestRun
Creating chain from scratch with Chain ID: ldsGNu
Chain built successfully 10
Loading chain from existing directory: /tmp/TestRun159058418/001/testnode-ldsGNu
Starting from height 10
Chain built successfully 20
    testing.go:1232: TempDir RemoveAll cleanup: unlinkat /tmp/TestRun159058418/001/testnode-ldsGNu/config: directory not empty
--- FAIL: TestRun (44.41s)
FAIL
FAIL	github.com/celestiaorg/celestia-app/v4/tools/chainbuilder	45.553s
```

--- a/tools/chainbuilder/integration_test.go
+++ b/tools/chainbuilder/integration_test.go
@@ -59,6 +59,7 @@ func TestRun(t *testing.T) {
 
 	appDB, err := tmdbm.NewDB("application", tmdbm.GoLevelDBBackend, tmCfg.DBDir())
 	require.NoError(t, err)
+	t.Cleanup(func() { _ = appDB.Close() })
 
 	app := app.New(
 		log.NewNopLogger(),

